kms++/src/videomode.cpp

```diff
@@ -110,18 +110,111 @@ static char sync_to_char(SyncPolarity pol)
 	}
 }
 
+#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
+
+template<typename T>
+std::string join(const T& values, const std::string& delim)
+{
+	std::ostringstream ss;
+	for (const auto& v : values) {
+		if (&v != &values[0])
+			ss << delim;
+		ss << v;
+	}
+	return ss.str();
+}
+
+static const char *mode_type_names[] = {
+	// the first 3 are deprecated so don't care about a short name
+	"builtin", // deprecated
+	"clock_c", // deprecated
+	"crtc_c", // deprecated
+	"P", // "preferred",
+	"default", // deprecated,
+	"U", // "userdef",
+	"D", // "driver",
+};
+
+static const char *mode_flag_names[] = {
+	// the first 5 flags are displayed elsewhere
+	NULL, // "phsync",
+	NULL, // "nhsync",
+	NULL, // "pvsync",
+	NULL, // "nvsync",
+	NULL, // "interlace",
+	"dblscan",
+	"csync",
+	"pcsync",
+	"ncsync",
+	"hskew",
+	"bcast", // deprecated
+	"pixmux", // deprecated
+	"2x", // "dblclk",
+	"clkdiv2",
+};
+
+static const char *mode_3d_names[] = {
+	NULL, "3dfp", "3dfa", "3dla", "3dsbs", "3dldepth", "3dgfx", "3dtab", "3dsbs",
+};
+
+static const char *mode_aspect_names[] = {
+	NULL, "4:3", "16:9", "64:27", "256:135",
+};
+
+const string mode_type_str(uint32_t inval) {
```

Comment:
"inval" means "invalid" =). Maybe "val" as the parameter name, and "v" as the local variable.